nodejs/lancedb/embedding/registry.ts

```diff
@@ -69,18 +65,25 @@ export class EmbeddingFunctionRegistry {
    * Fetch an embedding function by name
    * @param name The name of the function
    */
-  get<T extends EmbeddingFunction<unknown> = EmbeddingFunction>(
-    name: string,
-  ): EmbeddingFunctionCreate<T> | undefined {
+  get<T extends EmbeddingFunction<unknown>, Name extends string = "">(
+    name: Name extends "openai" ? "openai" : string,
+  ): Name extends "openai"
+    ? EmbeddingFunctionCreate<OpenAIEmbeddingFunction>
+    : EmbeddingFunctionCreate<T> | undefined {
```

Comment:
😵 Could you leave a comment describing what this means?

Reply:
TS type system is pretty ugly...

This makes it so that you can use string constants as "types", or use an explicitly supplied type
so 

`registry.get("openai") -> EmbeddingFunctionCreate<OpenAIEmbeddingFunction>`


`registry.get<MyCustomEmbeddingFunction>("my_func") -> EmbeddingFunctionCreate<MyCustomEmbeddingFunction> | undefined`

the reason this is important is that we always know our built in functions are defined so the user isnt forced to do a non null/undefined assertion

```ts
// before
const openai: OpenAIEmbeddingFunction | undefined = registry.get("openai")?.create();

// after
const openai: OpenAIEmbeddingFunction = registry.get("openai").create()

```

I'll also add this to the source code.